### ISSUE TITLE
[Feature Request]: Music management oriented feautres

### ISSUE THREAD

**Issue by TheElectroclassics** [TagStudioDev/TagStudio#523]
### Checklist

- [X] I am using an up-to-date version.
- [X] I have read the [documentation](https://github.com/TagStudioDev/TagStudio/blob/main/docs/index.md).
- [X] I have searched existing [issues](https://github.com/TagStudioDev/TagStudio/issues).

### Description

Full request
I'm looking to use this app to manage my music library, so I want to have a list view because the app only shows thumbnails.
<img width="1840" alt="Screenshot 2024-09-24 at 22 10 04" src="https://github.com/user-attachments/assets/7f38abe6-5846-4985-b1c7-ec52ce5d1b21">
Unfortunately, it appears it's only showing album art, so having a list and custom columns, like the DJ program Serato would be super useful. (here's a screenshot)
![Screenshot 2024-09-24 at 21 58 50](https://github.com/user-attachments/assets/939ffbee-af84-47d1-b645-77dde930635b)
Also, reading Serato documentation, it appears that they read ID3 tags from audio files. Maybe reading tags from AIFF and FLAC files can also help with this.
Lastly, the search bar on TagStudio seems to only search by tag implemented within the Library and not file names, which, as you can guess, is less than ideal.

Thank you for this project, I've been looking to manage by Library without needing to use iTunes (which no longer exists and now is the Music app for Mac), or Beatport's old library managing app. As you can see, my library is an absolute mess and I don't really want to manage it with the basic Windows Explorer or, in my case, Finder. The experience is subpar, but watching both YouTube videos on TagStudio, it looks like this app will solve my problem. Once again, thank you very much, I'll follow the development very closely.

(:

### Solution

TL;DR:
1. List View.
2. Custom Columns.
3. ID3 tag support.
4. Support for filename search.

**Comment by CyanVoxel** [comment]
- Audio playback added in #576
- Better filename search added in #575 and #582
